samples/embedded-auth-with-sdk/src/main/java/com/okta/spring/example/controllers/HomeController.java

```diff
@@ -176,8 +180,36 @@ public ModelAndView displayForgotPasswordPage(final HttpSession session) {
      */
     @GetMapping("/register")
     public ModelAndView displayRegisterPage(final HttpSession session) {
-        begin(session);
-        return new ModelAndView("register");
+        AuthenticationResponse authenticationResponse = begin(session);
+
+        authenticationResponse =
+                authenticationWrapper.fetchSignUpFormValues(authenticationResponse.getProceedContext());
+
+        ModelAndView modelAndView = new ModelAndView("register");
+
+        List<FormValue> userProfileAttributes = new LinkedList<>();
+        FormValue userProfileFormValue = null;
+
+        for (FormValue formValue: authenticationResponse.getFormValues()) {
+            if (formValue.getName().contentEquals("userProfile")) {
+                userProfileFormValue = formValue;
+            }
+        }
+
+        if (userProfileFormValue == null) {
+            displayErrorPage();
+        }
+
+        for (FormValue value: userProfileFormValue.form().getValue()) {
+            //Build the user profile
+            userProfileAttributes.add(value);
+        }
```

Comment:
Using `Optional<FormValue>` + one of `LinkedList` constructors this could be simplified to:

```java
List<FormValue> userProfileAttributes =
                new LinkedList<>(Arrays.asList(userProfileFormValue.get().form().getValue()));
```